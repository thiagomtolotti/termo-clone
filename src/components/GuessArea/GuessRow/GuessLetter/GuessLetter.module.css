.guess {
  width: 72px;
  aspect-ratio: 1;

  /* background: red; */
  color: white;
  font-weight: 600;
}

.guess.active {
  border-bottom-width: 11px !important;
}

.guess.correct {
  background-color: var(--green) !important;
}

.guess.misplaced {
  background-color: var(--yellow) !important;
}
