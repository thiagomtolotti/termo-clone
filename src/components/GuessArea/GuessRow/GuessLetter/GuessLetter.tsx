import styles from "./GuessLetter.module.css";

interface GuessLetterProps {
  children: React.ReactNode;
  active?: boolean;
  isCorrectOrMisplaced?: "correct" | "misplaced";
}

export const GuessLetter = ({
  active = false,
  children,
  isCorrectOrMisplaced,
}: GuessLetterProps) => {
  return (
    <div
      role="guess"
      className={`${styles.guess} ${active ? styles.active : ""} ${
        isCorrectOrMisplaced ? styles[isCorrectOrMisplaced] : ""
      }`}
    >
      {children}
    </div>
  );
};
