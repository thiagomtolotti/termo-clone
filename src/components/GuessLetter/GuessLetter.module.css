.guess {
  width: 72px;
  aspect-ratio: 1;

  /* background: red; */
  color: white;
  font-weight: 600;
}

.guess.active {
  border-bottom-width: 11px !important;
}
