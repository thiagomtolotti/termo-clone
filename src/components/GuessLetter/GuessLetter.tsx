import styles from "./GuessLetter.module.css";

interface GuessLetterProps {
  active?: boolean;
  children: React.ReactNode;
}

export const GuessLetter = ({ active = false, children }: GuessLetterProps) => {
  return (
    <div
      role="guess"
      className={`${styles.guess} ${active ? styles.active : ""}`}
    >
      {children}
    </div>
  );
};
