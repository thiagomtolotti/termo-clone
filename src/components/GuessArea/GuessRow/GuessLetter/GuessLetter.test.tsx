import { render, screen } from "@testing-library/react";
import { GuessLetter } from "./GuessLetter";

describe("GuessLetter component", () => {
  it("Should render correctly", () => {
    const { container } = render(<GuessLetter>T</GuessLetter>);

    expect(container).toMatchSnapshot();
  });

  it("Should render correctly if it's active", () => {
    const { container } = render(<GuessLetter active>T</GuessLetter>);

    expect(container).toMatchSnapshot();

    const guessElement = screen.getByRole("guess");
    expect(guessElement.classList.contains("active")).toBe(true);
  });
});
