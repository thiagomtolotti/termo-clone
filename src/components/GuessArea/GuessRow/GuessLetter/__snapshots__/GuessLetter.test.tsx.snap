// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuessLetter component Should render correctly 1`] = `
<div>
  <div
    class="guess "
    role="guess"
  >
    T
  </div>
</div>
`;

exports[`GuessLetter component Should render correctly if it's active 1`] = `
<div>
  <div
    class="guess active"
    role="guess"
  >
    T
  </div>
</div>
`;
